File: fastApiProject/entity/KnowledgeFile.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime


class KnowledgeFile(BaseModel):
    """知识文件表"""
    id: Optional[int] = Field(None, description="文件ID")
    knowledge_base_id: int = Field(..., description="所属知识库ID")
    file_name: str = Field(..., description="文件名", max_length=512)
    file_size: int = Field(..., description="文件大小（字节）")
    file_type: Optional[str] = Field(None, description="文件类型", max_length=100)
    file_hash: str = Field(..., description="文件哈希值", max_length=64)
    indexing_method: str = Field("semantic", description="索引方法", max_length=50)
    status: str = Field("pending", description="状态", max_length=20)
    error_message: Optional[str] = Field(None, description="错误信息")
    file_metadata: Dict[str, Any] = Field(default_factory=dict, description="元数据")
    created_at: Optional[datetime] = Field(None, description="创建时间")
    updated_at: Optional[datetime] = Field(None, description="更新时间")
    is_deleted: int = Field(0, description="逻辑删除标志:0-未删除,1-已删除")

    class Config:
        from_attributes = True

File: blog-agent/agent/src/main/java/com/xiafan/agent/entity/KnowledgeFile.java
package com.xiafan.agent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.xiafan.agent.repository.PostgresJsonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

/** knowledge_files */
@TableName(value = "knowledge_files", autoResultMap = true)
@Data
public class KnowledgeFile {
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    @TableField("knowledge_base_id")
    private Integer knowledgeBaseId;
    @TableField("file_name")
    private String fileName;
    @TableField("file_size")
    private Integer fileSize;
    @TableField("file_type")
    private String fileType;
    @TableField("file_hash")
    private String fileHash;
    @TableField("indexing_method")
    private String indexingMethod = "semantic";
    @TableField("status")
    private String status = "pending";
    @TableField("error_message")
    private String errorMessage;
    @TableField(value = "file_metadata", typeHandler = PostgresJsonTypeHandler.class)
    private Map<String, Object> fileMetadata = new HashMap<>();
    @TableField("created_at")
    private LocalDateTime createdAt;
    @TableField("updated_at")
    private LocalDateTime updatedAt;
    @TableField("is_deleted")
    private Integer isDeleted = 0;
}
